lib/dom.js

```diff
@@ -2400,42 +2427,74 @@ Text.prototype = {
 	},
 };
 _extends(Text, CharacterData);
-function Comment() {}
+function Comment(symbol) {
+	if (symbol !== INTERNAL_SYMBOL) {
+		throw new Error('Illegal constructor');
+	}
+}
 Comment.prototype = {
 	nodeName: '#comment',
 	nodeType: COMMENT_NODE,
 };
 _extends(Comment, CharacterData);
 
-function CDATASection() {}
+function CDATASection(symbol) {
+	if (symbol !== INTERNAL_SYMBOL) {
+		throw new Error('Illegal constructor');
+	}
+}
 CDATASection.prototype = {
 	nodeName: '#cdata-section',
 	nodeType: CDATA_SECTION_NODE,
 };
 _extends(CDATASection, CharacterData);
 
-function DocumentType() {}
+function DocumentType(symbol) {
+	if (symbol !== INTERNAL_SYMBOL) {
+		throw new Error('Illegal constructor');
+	}
+}
 DocumentType.prototype.nodeType = DOCUMENT_TYPE_NODE;
 _extends(DocumentType, Node);
 
-function Notation() {}
+function Notation(symbol) {
+	if (symbol !== INTERNAL_SYMBOL) {
+		throw new Error('Illegal constructor');
+	}
+}
 Notation.prototype.nodeType = NOTATION_NODE;
 _extends(Notation, Node);
 
-function Entity() {}
+function Entity(symbol) {
+	if (symbol !== INTERNAL_SYMBOL) {
+		throw new Error('Illegal constructor');
+	}
+}
 Entity.prototype.nodeType = ENTITY_NODE;
 _extends(Entity, Node);
 
-function EntityReference() {}
+function EntityReference(symbol) {
+	if (symbol !== INTERNAL_SYMBOL) {
+		throw new Error('Illegal constructor');
+	}
+}
 EntityReference.prototype.nodeType = ENTITY_REFERENCE_NODE;
 _extends(EntityReference, Node);
 
-function DocumentFragment() {}
+function DocumentFragment(symbol) {
+	if (symbol !== INTERNAL_SYMBOL) {
+		throw new Error('Illegal constructor');
+	}
```

Comment:
I think it's worth to extract this into a function on the module level which we pass the symbol to.